digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/ProjectDAO.scala

```diff
@@ -284,76 +285,85 @@ object ProjectDAO {
       val calibrationPointsSource = CalibrationPointSource.apply(r.nextIntOption().getOrElse(99))
 
       ProjectLink(projectLinkId, roadNumber, roadPartNumber, trackCode, discontinuityType, startAddrM, endAddrM, startDate, endDate,
-        modifiedBy, linkId, startMValue, endMValue, sideCode, CalibrationPointsUtils.toProjectLinkCalibrationPointsWithSourceInfo(calibrationPoints, calibrationPointsSource), NoFloating, parseStringGeometry(geom.getOrElse("")), projectId,
+        modifiedBy, linkId, startMValue, endMValue, sideCode, CalibrationPointsUtils.toProjectLinkCalibrationPointsWithSourceInfo(calibrationPoints, calibrationPointsSource), NoFloating, OracleDatabase.load(geom), projectId,
         status, roadType, source, length, roadAddressId, ely, reversed, connectedLinkId, geometryTimeStamp, roadName = Some(roadName),
         roadAddressLength = roadAddressEndAddrM.map(endAddr => endAddr - roadAddressStartAddrM.getOrElse(0L)),
         roadAddressStartAddrM = roadAddressStartAddrM, roadAddressEndAddrM = roadAddressEndAddrM, roadAddressTrack = roadAddressTrack,
         roadAddressRoadNumber = roadAddressRoadNumber, roadAddressRoadPart = roadAddressRoadPart)
     }
   }
 
-
-  private def parseStringGeometry(geomString: String): Seq[Point] = {
-    if (geomString.nonEmpty)
-      toGeometry(geomString)
-    else
-      Seq()
-  }
-
   private def listQuery(query: String) = {
     Q.queryNA[ProjectLink](query).iterator.toSeq
   }
 
+  //TODO delete when there is no longer needed. This is now need to script where we migrate PROJECT_LINK .GEOMETRY_STRING to .GEOMETRY
+  def updateGeometryStringToSdo(id: Long, geometry: Seq[Point]): Unit = {
```

Comment:
Now I think is no longer needed 🥇

Reply:
This one will be used to the batch to migrate PROJETC_LINK(_HISTORY) Geometry (VARCHAR (4000 BYTE)) to Geometry (SDO_GEOMETRY). After that the batch and its code will be deprecated.